src/main/scala/cromwell/util/docker/DockerHash.scala

```diff
@@ -0,0 +1,110 @@
+package cromwell.util.docker
+
+import javax.xml.bind.DatatypeConverter
+
+import cromwell.util.StringUtil._
+import cromwell.util.TryUtil
+
+import scala.util.{Failure, Success, Try}
+import scalaz.Scalaz._
+import scalaz.{Failure => FailureZ, Success => SuccessZ, _}
+
+case class DockerHash(hashType: String, hashString: String) {
+  val digest = s"$hashType:$hashString"
+}
+
+object DockerHash {
+  /**
+    * Creates a unique hash from a sequence of hashes.
+    * If there is a single hash, returns just the single hash string.
+    * If there is more than one hash, concatenates the hash strings together, and then md5s the resulting string.
+    * Returns a failure if there are zero hashes, or if there are more than one type of hash type.
+    *
+    * @param hashCollectionType A description of the collection.
+    * @param dockerHashes The sequence of hashes to rehash.
+    * @return A single docker hash.
+    */
+  def fromSeq(hashCollectionType: String, dockerHashes: Seq[DockerHash]): Try[DockerHash] = {
+    if (dockerHashes.isEmpty) {
```

Comment:
Could you do this via pattern match? Since you're using Seq I suppose it'd end up being kinda ugly